scripts/results_verification.py

```diff
@@ -0,0 +1,98 @@
+from pathlib import Path
+from argparse import ArgumentParser
+import xml.etree.ElementTree as ET
+
+
+def check_file_size(file_path, max_size=5):
+    """
+    Check if the file size of a given file is smaller than max_size.
+
+    :param file_path: path of junit xml report from twister to be checked
+    :param max_size: max size allowed for the given file
+    """
+    file_size = file_path.stat().st_size / 1024 / 1024
+    return bool(file_size <= max_size)
+
+
+def check_name(file_path):
+    """
+    Check if the name of the file corresponds with the platform name in the report
+
+    :param file_path: path of junit xml report from twister to be checked
+    """
+    platform = file_path.name[:-4]
+    summary = ET.parse(file_path).getroot()[0]
+    name_in_report = summary.attrib['name']
+    return bool(platform == name_in_report)
+
+
+def check_attribute_value(file_path=None, item=None, max_value=None):
+    """
+    Check if the value of a given attribute is smaller than max_count. Attribute can be e.g. 'errors' or 'failures'
+
+    :param file_path: path of junit xml report from twister to be checked
+    :param item: attribute to be verified
+    :param max_value: max value of the given attribute
+    """
+    summary = ET.parse(file_path).getroot()[0]
+    item_count = int(summary.attrib[item])
+    return bool(item_count <= max_value)
+
+
+def check_version(file_path=None, version=None):
+    """
+    Check if zephyr's version given in the report match the expected one
+
+    :param file_path: junit xml report from twister
+    :param version: expected version of zephyr
+    """
+    testsuite = ET.parse(file_path).getroot()[0]
+    properties = testsuite[0]
+    for property in properties:
+        if property.attrib['name'] == "version":
+            return bool(property.attrib['value'] == version)
+
+    print("Version not found.")
+    return False
+
+
+def parse_args():
+    """Parse and return required limits from arguments"""
+    argpar = ArgumentParser(description='Verifies if a given report fulfils requirements before its publishing')
+    argpar.add_argument('-P', '--path', required=True, type=str, help='Path to the report which will be verified')
+    argpar.add_argument('-V', '--version', required=True, type=str, help='Version to be verified')
```

Comment:
--version is usually used to print out the version of the script itself, pick something else please

Reply:
Changed to '-Z'/ '--zephyr'